pkg/cmd/project/link/link.go

```diff
@@ -0,0 +1,195 @@
+			if err != nil {
+				return err
+			}
+
+			if len(args) == 1 {
+				num, err := strconv.ParseInt(args[0], 10, 32)
+				if err != nil {
+					return cmdutil.FlagErrorf("invalid number: %v", args[0])
+				}
+				opts.number = int32(num)
+			}
+
+			config := linkConfig{
+				httpClient: f.HttpClient,
+				client:     client,
+				opts:       opts,
+				io:         f.IOStreams,
+			}
+
+			if config.opts.repo != "" && config.opts.team != "" {
+				return fmt.Errorf("specify only one repo or team")
+			} else if config.opts.repo == "" && config.opts.team == "" {
+				return fmt.Errorf("specify at least one repo or team")
+			}
+
+			// allow testing of the command without actually running it
+			if runF != nil {
+				return runF(config)
+			}
+			return runLink(config)
+		},
+	}
+
+	linkCmd.Flags().StringVar(&opts.owner, "owner", "", "Login of the owner. Use \"@me\" for the current user.")
+	linkCmd.Flags().StringVarP(&opts.repo, "repo", "R", "", "The repository to be linked to this project")
+	linkCmd.Flags().StringVarP(&opts.team, "team", "T", "", "The team to be linked to this project")
+	cmdutil.AddFormatFlags(linkCmd, &opts.exporter)
+
+	return linkCmd
+}
+
+func runLink(config linkConfig) error {
+	canPrompt := config.io.CanPrompt()
+	owner, err := config.client.NewOwner(canPrompt, config.opts.owner)
+	if err != nil {
+		return err
+	}
+
+	project, err := config.client.NewProject(canPrompt, owner, config.opts.number, false)
+	if err != nil {
+		return err
+	}
+	config.opts.projectID = project.ID
+
+	httpClient, err := config.httpClient()
+	if err != nil {
+		return err
+	}
+	c := api.NewClientFromHTTP(httpClient)
+
+	if config.opts.repo != "" {
+		repo, err := api.GitHubRepo(c, ghrepo.New(owner.Login, config.opts.repo))
+		if err != nil {
+			return err
+		}
+		config.opts.repoID = repo.ID
+
+		query, variable := linkRepoArgs(config)
+		err = config.client.Mutate("LinkProjectV2ToRepository", query, variable)
+		if err != nil {
+			return err
+		}
+
+		if config.opts.exporter != nil {
+			return config.opts.exporter.Write(config.io, query.LinkProjectV2ToRepository.Repository)
+		}
+		return printResults(config, query.LinkProjectV2ToRepository.Repository.URL)
+
+	} else if config.opts.team != "" {
+		teams, err := api.OrganizationTeams(c, ghrepo.New(owner.Login, ""))
+		if err != nil {
+			return err
+		}
+		for _, team := range teams {
+			if team.Slug == config.opts.team {
+				config.opts.teamID = team.ID
+				break
+			}
+		}
+		if config.opts.teamID == "" {
+			return fmt.Errorf("can't find team %s", config.opts.team)
+		}
+
+		query, variable := linkTeamArgs(config)
+		err = config.client.Mutate("LinkProjectV2ToTeam", query, variable)
+		if err != nil {
+			return err
+		}
+
+		if config.opts.exporter != nil {
+			return config.opts.exporter.Write(config.io, query.LinkProjectV2ToTeam.Team)
+		}
+		return printResults(config, query.LinkProjectV2ToTeam.Team.URL)
+	}
+	return fmt.Errorf("specify at least one repo or team")
```

Comment:
If we are handling this in `RunE`, then we can just return nil here:

```suggestion
	return nil
```